### PR TITLE
add blacklist functionality

### PR DESCRIPTION
**Pull request by Coiling-Dragon**
**Detailed description**:
Adds transaction block functionality for given addresses 

**Which issue(s) this PR fixes**:
Fixes #<issue number>

**Special notes for your reviewer**:

**Checklist**
- [X] Documentation added
- [x] Tests updated

### CODE REVIEW

File: app/helper/blacklist/blacklist.go [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+			return true
+		}
+	}
+	return false
+}
+
+// Checks if the transaction contains any blacklisted accounts in any transfer
+func CheckTxForBlacklistedAccounts(blackListedAccounts []string, tx transaction.Transaction) error {
+	for i := range tx.Transfers {
+		fmt.Printf("\nAcc: %v", tx.Transfers[i].Account)
```
Comment (rokn) [inline app/helper/blacklist/blacklist.go:36]: Can we remove the 3 prints here

File: app/helper/blacklist/helper_test.go [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+)
+
+func Test_IsBlacklistedAccount(t *testing.T) {
+	blacklist := []string{"0x000000", "0x000001"}
+	assert.True(t, IsBlacklistedAccount(blacklist, "0x000000"))
+	assert.False(t, IsBlacklistedAccount(blacklist, "0x000002"))
+}
+
+func Test_CheckTxForBlacklistedAccounts(t *testing.T) {
+	// tx :=
```
Comment (rokn) [inline app/helper/blacklist/helper_test.go:32]: nit: comment

File: app/helper/blacklist/helper_test.go [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+	"testing"
+)
+
+func Test_IsBlacklistedAccount(t *testing.T) {
+	blacklist := []string{"0x000000", "0x000001"}
+	assert.True(t, IsBlacklistedAccount(blacklist, "0x000000"))
+	assert.False(t, IsBlacklistedAccount(blacklist, "0x000002"))
+}
+
+func Test_CheckTxForBlacklistedAccounts(t *testing.T) {
```
Comment (rokn) [inline app/helper/blacklist/helper_test.go:31]: I'd split this test into multiple ones, testing each of the for loops and their happy and not happy paths. I know it's a bit tedious but that way we are sure there won't be any sdk changes that would alter the functionality... and we get our coverage up ;) And for the splitting I'd say it's easier to pinpoint which check is failing when there is one per test

File: app/process/watcher/evm/watcher.go [hunk 4]
```diff
@@ -178,20 +181,23 @@ func NewWatcher(
...
-		pricingService:    pricingService,
-		evmClient:         evmClient,
-		logger:            c.GetLoggerFor(fmt.Sprintf("EVM Router Watcher [%s]", dbIdentifier)),
-		assetsService:     assetsService,
-		targetBlock:       targetBlock,
-		validator:         validator,
-		sleepDuration:     pollingInterval,
-		filterConfig:      filterConfig,
-	}
+	istance := &Watcher{
```
Comment (rokn) [inline app/process/watcher/evm/watcher.go:184]: Why do we need the variable?

File: app/process/watcher/evm/watcher.go [hunk 5]
```diff
@@ -246,6 +252,25 @@ func (ew Watcher) beginWatching(queue qi.Queue) {
 	}
 }
 
+func (ew Watcher) CheckBlacklistedOriginator(hash common.Hash) (*string, error) {
+	tx, err := ew.evmClient.RetryTransactionByHash(hash)
+	if err != nil {
+		err := fmt.Errorf("[%s] - Failed to get transaction by hash. Error: [%s]", hash, err)
+		return nil, err
+	}
```
Comment (rokn) [inline app/process/watcher/evm/watcher.go:260]: nit: some blank lines after the if's wouldn't hurt

File: app/process/watcher/transfer/watcher_test.go [hunk 6]
```diff
@@ -188,6 +194,69 @@ func Test_ProcessTransaction_SanityCheckTransfer_Fails(t *testing.T) {
...
+
+	w := initializeWatcher()
+	mocks.MHederaMirrorClient.On("GetSuccessfulTransaction", tx_blacklist.TransactionID).Return(tx_blacklist, nil)
+
+	w.processTransaction(tx_blacklist.TransactionID, mocks.MQueue)
+
+	mocks.MQueue.AssertNotCalled(t, "Push", mock.Anything)
+}
+
+func Test_ProcessHbarransaction_Blacklist_Fails(t *testing.T) {
```
Comment (rokn) [inline app/process/watcher/transfer/watcher_test.go:234]:
```suggestion
func Test_ProcessHbarTransaction_Blacklist_Fails(t *testing.T) {
```

File: bootstrap/server.go [hunk 7]
```diff
@@ -168,6 +168,7 @@ func registerEvmClients(server *server.Server, services *Services, repositories
 		// EVM networks might be the same, a concatenation between
 		// <chain-id>-<contract-address> removes possible duplication.
 		dbIdentifier := fmt.Sprintf("%d-%s", chain, contractService.Address().String())
+		black_listed := configuration.Bridge.BlackListedAccounts
```
Comment (rokn) [inline bootstrap/server.go:171]: nit: 
```suggestion
		blacklisted := configuration.Bridge.BlackListedAccounts
```

File: bootstrap/watchers.go [hunk 8]
```diff
@@ -40,6 +41,7 @@ func createTransferWatcher(configuration *config.Config,
 	pricingService service.Pricing,
 ) *tw.Watcher {
 	account := configuration.Bridge.Hedera.BridgeAccount
+	black_listed_accounts := configuration.Bridge.BlackListedAccounts
```
Comment (rokn) [inline bootstrap/watchers.go:44]:
```suggestion
	blacklisted_accounts := configuration.Bridge.BlackListedAccounts
```

File: config/bridge.go [hunk 9]
```diff
@@ -30,13 +30,14 @@ import (
...
-	MinAmounts        map[uint64]map[string]*big.Int
-	MonitoredAccounts map[string]string
+	TopicId             string
+	Hedera              *BridgeHedera
+	EVMs                map[uint64]BridgeEvm
+	CoinMarketCapIds    map[uint64]map[string]string
+	CoinGeckoIds        map[uint64]map[string]string
+	MinAmounts          map[uint64]map[string]*big.Int
+	MonitoredAccounts   map[string]string
+	BlackListedAccounts []string
```
Comment (rokn) [inline config/bridge.go:40]:
```suggestion
	BlacklistedAccounts []string
```

File: config/parser/bridge.go [hunk 10]
```diff
@@ -25,13 +25,14 @@ import (
...
-	Networks          map[uint64]*Network `yaml:"networks,omitempty" json:"networks,omitempty"`
-	MonitoredAccounts map[string]string   `yaml:"monitored_accounts,omitempty" json:"monitoredAccounts,omitempty"`
+	UseLocalConfig      bool                `yaml:"use_local_config,omitempty" json:"useLocalConfig,omitempty"`
+	ConfigTopicId       string              `yaml:"config_topic_id,omitempty" json:"configTopicId,omitempty"`
+	FeePolicyTopicId    string              `yaml:"fee_policy_topic_id,omitempty" json:"feePolicyTopicId,omitempty"`
+	PollingInterval     time.Duration       `yaml:"polling_interval,omitempty" json:"pollingInterval,omitempty"`
+	TopicId             string              `yaml:"topic_id,omitempty" json:"topicId,omitempty"`
+	Networks            map[uint64]*Network `yaml:"networks,omitempty" json:"networks,omitempty"`
+	MonitoredAccounts   map[string]string   `yaml:"monitored_accounts,omitempty" json:"monitoredAccounts,omitempty"`
+	BlackListedAccounts []string            `yaml:"black_list,omitempty" json:"blackListedAccounts,omitempty"`
```
Comment (rokn) [inline config/parser/bridge.go:35]:
```suggestion
	BlacklistedAccounts []string            `yaml:"blacklist,omitempty" json:"blacklistedAccounts,omitempty"`
```

File: test/mocks/client/evm_client_mock.go [hunk 11]
```diff
@@ -197,6 +197,6 @@ func (m *MockEVM) WaitForTransactionReceipt(hash common.Hash) (txReceipt *types.
 }
 
 func (m *MockEVM) RetryTransactionByHash(hash common.Hash) (*types.Transaction, error) {
-	//TODO implement me
-	panic("implement me")
-}
+	args := m.Called(hash)
+	return args.Get(0).(*types.Transaction), nil
+}
```
Comment (rokn) [inline test/mocks/client/evm_client_mock.go:202]: nit: no new line

File: app/helper/blacklist/blacklist.go (hunk 1 above)
Comment (Coiling-Dragon) [reply app/helper/blacklist/blacklist.go:36]: Yes sorry, forgot to delete after testing.

File: app/process/watcher/evm/watcher.go (hunk 4 above)
Comment (Coiling-Dragon) [reply app/process/watcher/evm/watcher.go:184]: It is easier to debug this way. Should I remove it?

Comment (Coiling-Dragon) [reply app/process/watcher/evm/watcher.go:184]: removed

**Review by rokn — DISMISSED** [review]

**Review by rokn — APPROVED** [review]

**Pull request merged**